### PR TITLE
feat(2fa):  add input files components

### PR DESCRIPTION
**Pull request by oalkabouss**
ref: MANAGER-14223

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/acquisition-storage-2fa`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix # MANAGER-14223
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Add file selection component

### CODE REVIEW

File: packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInput.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  className?: string;
+};
+
+export const FileInput: FunctionComponent<Props> = ({ className }) => {
+  const context = useContext(FileInputContext);
+  if (!context) {
+    throw Error(
+      'The component <FileInput /> must be a child of FileInputContainer',
+    );
+  }
```
Comment (chipp972) [inline packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInput.tsx:23]: Is it necessary ? `useContext` won't throw before if there is no context found ?

File: packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInput.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    const fileList = event.target.files;
+    if (!fileList) return;
+    const newFiles: File[] = multiple
+      ? [...value, ...Array.from(fileList)]
+      : Array.from(fileList);
+
+    if (onChange) {
+      onChange({ files: newFiles.slice(0, maxFiles), e: event });
+    }
+    fileInputRef.current.value = '';
```
Comment (chipp972) [inline packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInput.tsx:40]:
```suggestion
    onChange?.({ files: newFiles.slice(0, maxFiles), e: event });
    fileInputRef.current.value = '';
```

File: packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInputErrorMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        });
+      }
+
+      if (file.size > maxSize) {
+        error = t('account-disable-2fa-file-input-size-file-error');
+      }
+    });
+
+    setErrorMessage(error);
+  }, [files]);
```
Comment (chipp972) [inline packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInputErrorMessage.tsx:53]: You only keep the last error you found ?

File: packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInputList.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+  const { value, multiple, onChange } = context;
+
+  const handleDeleteFile = (
+    e: MouseEvent<HTMLOsdsIconElement, globalThis.MouseEvent>,
+    file: File,
+  ) => {
+    if (onChange) {
+      onChange({ files: value.filter((f) => f.name !== file.name), e });
+    }
```
Comment (chipp972) [inline packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInputList.tsx:69]:
```suggestion
    onChange?.({ files: value.filter((f) => f.name !== file.name), e });
```

File: packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInput.tsx (hunk 1 above)
Comment (oalkabouss) [reply packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInput.tsx:23]: Yeah, useContext doesn't throw an exception if the component is outside of the provider, useContext simply returns an undefined context.

**Review by chipp972 — APPROVED** [review]

File: packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInputList.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            : ODS_THEME_COLOR_INTENT.primary
+        }
+        className="ml-auto cursor-pointer"
+        size={ODS_ICON_SIZE.sm}
+        onClick={(e) => deleteFile(e, file)}
+      />
+    </div>
+  );
+};
+
```
Comment (MaximeBajeux) [inline packages/manager/apps/account-disable-2fa/src/components/FileInput/FileInputList.tsx:99]: Should we not split the file in two with each component in his own file ?

**Review by MaximeBajeux — APPROVED** [review]

**Pull request merged**